src/ucp/am/rndv.c

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2022, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "ucp_am.inl"
+
+#include <ucp/proto/proto_single.inl>
+#include <ucp/rndv/proto_rndv.inl>
+
+
+static size_t ucp_am_rndv_rts_pack(void *dest, void *arg)
+{
+    ucp_request_t *req          = arg;
+    ucp_rndv_rts_hdr_t *rts_hdr = dest;
+    size_t rts_size, max_bcopy, total_size;
+
+    ucp_am_fill_header(ucp_am_hdr_from_rts(rts_hdr), req);
+    rts_hdr->opcode = UCP_RNDV_RTS_AM;
+    rts_size        = ucp_proto_rndv_rts_pack(req, rts_hdr, sizeof(*rts_hdr));
+    if (req->send.msg_proto.am.header_length == 0) {
+        return rts_size;
+    }
+
+    total_size = rts_size + req->send.msg_proto.am.header_length;
+    max_bcopy  = ucp_ep_get_max_bcopy(req->send.ep, req->send.lane);
+    if (ucs_unlikely(total_size > max_bcopy)) {
+        ucs_fatal("RTS is too big %lu, max %lu", total_size, max_bcopy);
+    }
+
+    ucp_am_pack_user_header(UCS_PTR_BYTE_OFFSET(rts_hdr, rts_size), req);
+
+    return total_size;
+}
+
+UCS_PROFILE_FUNC(ucs_status_t, ucp_am_rndv_proto_progress, (self),
+                 uct_pending_req_t *self)
+{
+    ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
+    const ucp_proto_rndv_ctrl_priv_t *rpriv;
+    ucs_status_t status;
+    size_t max_rts_size;
+
+    rpriv  = req->send.proto_config->priv;
+    status = UCS_PROFILE_CALL(ucp_proto_rndv_rts_request_init, req);
+    if (status != UCS_OK) {
+        ucp_proto_request_abort(req, status);
+        return UCS_OK;
+    }
+
+    req->send.lane = rpriv->lane;
```

Comment:
this is already done by ucp_proto_single_status_handle, as part of ucp_proto_am_bcopy_single_progress